get-cabal-configuration.hs

```diff
@@ -248,10 +271,18 @@ getBuildDirectories tool pkgDesc cabalDir = do
         componentBuildDir componentName =
             buildDir </> componentName </> (componentName ++ "-tmp")
 
+        newComponentBuildDir :: String -> FilePath
+        newComponentBuildDir componentName =
+            cabalDir </> distDir'
+                     </> "build"
+                     </> componentName
+                     </> (componentName ++ "-tmp")
```

Comment:
I'm not sure that this is correct. According to https://cabal.readthedocs.io/en/latest/nix-local-build.html#where-are-my-build-products, path to component may look like `dist-newstyle/build/x86_64-linux/ghc-8.0.1/p-0.1/c/pexe`, but this code does not look like it cat produce that. Am I missing something?

I'm also aware that depending on, say, optimisation options the component build directory changes subtly by having additional `/o` somewhere in it. Are we handling changes like that too?

Reply:
I was already aware of that page but it's not matching directory layout I'm seeing empirically in my system. I checked it with the `lens` project which includes testing, executable and benchmark components.

Around a year ago I attempted to do what I'm doing now, and I remember seeing single letter directories per component in the directory layouts, but now, with Cabal-2.2.0.1 and ghc-8.4.3 I'm not seeing them, even if I try to build the components individually.

Are you getting the single letter characters in your system?

Reply:
Yes I ho get them if, for example, I specify `optimization: 2` in my `cabal.project`.